CONTRIBUTING.md

```diff
@@ -25,6 +25,20 @@ Pull Requests
 * **DO NOT** fix merge conflicts using a merge commit. Prefer `git rebase`.
 * **DO NOT** mix independent, unrelated changes in one PR. Separate real product/test code changes from larger code formatting/dead code removal changes. Separate unrelated fixes into separate PRs, especially if they are in different assemblies.
 
+Pull Request Checklist
+----------------------
+
+Please leverage this checklist as a reminder to address commonly occurring feedback when submitting a pull request to make sure your PR can be reviewed quickly:
+
+- [ ] Does the API design and implementation within your PR follow the [C++ Guidelines](https://azure.github.io/azure-sdk/cpp_introduction.html)?
+- [ ] If you are modifying header files, are the changes documented using doxygen style comments?
+- [ ] Have you added relevant unit tests to ensure CI will catch future regressions?
+- [ ] Have you reviewed your own PR to make sure there aren't any unintended changes or commits?
+- [ ] Does your PR have a descriptive title and description (with issue number), which explains why the change is being made?
+- [ ] Do you have any complex components that could use comments in source to explain (focusing on the "why")?
+- [ ] Are there any typos or spelling errors, especially in user-facing documentation?
+- [ ] Is the PR actually ready for review or is it work-in-progress (WIP) or an experiment? If so, can it start off as a draft PR?
```

Comment:
Didn't mention linters and auto code-formatters for now until we have those guards already in CI. At that point, we can add those to help contributors.